src/hubmap_translator.py

```diff
@@ -651,7 +663,16 @@ def generate_doc(self, entity, return_type):
                     if 'ingest_metadata' in entity:
                         ingest_metadata = entity['ingest_metadata']
                         if 'files' in ingest_metadata:
-                            entity['files'] = ingest_metadata['files']
+                            if ingest_metadata['files'].strip() == "":
```

Comment:
Is this the assumption that the `Dataset.ingest_metadata.files` field is ALWYAS either a list of dicts or empty string? Otherwise, could it be an empty list `[]` as well?

@DerekFurstPitt this is not a change request, it's more about confirming all the possible cases with Joel.

I also noticed that we have this existing block: https://github.com/hubmapconsortium/search-api/blob/Derek-Furst/expose-metadata/src/hubmap_translator.py#L692-L695 that removes the `metadata.files` after renaming from the `ingest_metadata`. I'm wondering if we should move this block right below your changes and keep using `ingest_metadata` since it's not yet renamed at that point when IF we move it there. Just make it more readable?

Reply:
@yuanzhou Updated and repushed